src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -1049,6 +1049,12 @@ public static function deserialize($varValue, $blnForceArray=false)
 			return $blnForceArray ? array() : '';
 		}
 
+		// Return if definitely not a serialized string
+		if (!preg_match('/^[CONRSabdiors]:[0-9]+:/', $varValue))
```

Comment:
I've checked the PHP C source and here's the result:

* `C` and `O` are for classes and objects
* `N` is `null`
* `R` and `r` are references
* `a` is an `array`
* `b` is a `boolean`
* `d` is a `double`
* `i` is an `integer`
* `s` is a `string`

I did not find `S` and `o`, what do they stand for?

Reply:
I don't know. I just know that they exist: https://github.com/php/php-src/blob/master/ext/standard/var_unserializer.c#L711-L722

Reply:
Okay, then please leave that reference as a comment in the source code so that we still know in the future. Otherwise this PR is just fine 👍

Reply:
Added in 9974fd7e02f836b93c9fed95f79ce14863d720cf. Still I would like to know what `S` and `o` stand for. 🤔 @ausi ?

Reply:
`S` is for escaped strings, e.g. `S:1:"\5A";` computes to `"Z"`.

`o` is for `stdClass` objects, e.g. `o:1:"s:3:"foo";s:3:"bar";}` computes to an object with a `foo` property.

Both do not get generated by `serialize()` (anymore) AFAIK.